retrodep/gosource.go

```diff
@@ -533,7 +533,11 @@ func (src GoSource) RepoPathForImportPath(importPath string) (*RepoPath, error)
 
 	// No replacement found, use the import pth as-is
 	r, err := vcs.RepoRootForImportPath(importPath, false)
-	if err != nil && strings.ContainsRune(importPath, '_') {
+
+	if err != nil {
+		if !strings.ContainsRune(importPath, '_') {
+			return nil, err
```

Comment:
Maybe we also want to return this if the later call (https://github.com/release-engineering/retrodep/pull/120/files#diff-1ccdbf0a977cf6929c983e45f2193e48R546) fails? e.g.
```
if nerr != nil {
    return nil, err
}
return &RepoPath{...}...
```

Reply:
> Maybe we also want to return this if the later call (https://github.com/release-engineering/retrodep/pull/120/files#diff-1ccdbf0a977cf6929c983e45f2193e48R546) fails? e.g.
> 
> ```
> if nerr != nil {
>     return nil, err
> }
> return &RepoPath{...}...
> ```

Changing to 
`		
	if err == nil {
		return &RepoPath{RepoRoot: *r}, err
	}

	if !strings.ContainsRune(importPath, '_') {
		return nil, err
	}
	// gopkg.in gives bad responses for paths like
	// gopkg.in/foo/bar.v2/_examples/chat1
	// because of the underscore. Remove it and try again.
	u := strings.Index(importPath, "_")
	importPath = path.Dir(importPath[:u])
	r, nerr := vcs.RepoRootForImportPath(importPath, false)
	if nerr != nil {
		return nil, err
	}
	return &RepoPath{RepoRoot: *r}, nil
`

Reply:
With the current change any error encountered on a package will retrodep to just exit.
Adding another change that will cause the error to be printed, but the rest of the packages be processed normally.
Let me know if that behavior is ok.